control/tap-ctl-ipc.c

```diff
@@ -61,6 +61,7 @@ tap_ctl_read_raw(int fd, void *buf, size_t size, struct timeval *timeout)
 	fd_set readfds;
 	size_t offset = 0;
 	int ret;
+	int err = 0;
 
 	while (offset < size) {
 		FD_ZERO(&readfds);
```

Comment:
This only watches one file, the zeroing could be moved out of the loop, no need to zero all 1024 on each iteration. Alternatively `poll` could be used instead of select if you want to watch just 1 file (select might still have to traverse the bitmap to find out what to watch).

Reply:
Probably too big a change to switch to poll, it might be reasonable in a future change but at the moment just trying to fix the misreporting of the time out so that the caller can make sensible decisions in the case of a time out.